File: packages/renderer-vue/playground/measurements/MeasurementLengthSumNode.ts
import { defineNode, NodeInterface } from "@baklavajs/core";
import { setType } from "@baklavajs/interface-types";
import { Measurement } from "../models/Measurement";
import { numberType, measurementListType } from "../interfaceTypes";

export default defineNode({
    type: "MeasurementLengthSumNode",
    title: "Length Sum",
    inputs: {
        measurements: () => new NodeInterface<Measurement[]>("Measurements", []).use(setType, measurementListType),
    },
    outputs: {
        lengthSum: () => new NodeInterface<number>("Length Sum", 0).use(setType, numberType),
    },
    calculate(inputs) {
        // Sum up all the lengths from the measurements array
        let lengthSum = 0;

        // inputs.measurements should be an array of measurements
        if (Array.isArray(inputs.measurements)) {
            for (const measurement of inputs.measurements) {
                if (measurement && typeof measurement.length === 'number') {
                    lengthSum += measurement.length;
                }
            }
        }

        return { lengthSum };
    },
});

File: BaklavaDependencyEngine/BaklavaDependencyEngine.Core/ExampleNodes/MeasurementLengthSumNode.cs
using BaklavaDependencyEngine.Core.Models;

namespace BaklavaDependencyEngine.Core.ExampleNodes
{
    public class MeasurementLengthSumNode : Node
    {
        public MeasurementLengthSumNode() : base("MeasurementLengthSumNode", "Length Sum")
        {
        }

        protected override void InitializeInterfaces()
        {
            AddInput("measurements", new NodeInterface("Measurements", new List<Measurement>()));
            AddOutput("lengthSum", new NodeInterface("Length Sum", 0.0, false));
        }

        public override async Task<Dictionary<string, object>> Calculate(
            Dictionary<string, object> inputs,
            CalculationContext context)
        {
            var measurements = inputs["measurements"] as List<Measurement> ?? new List<Measurement>();

            double lengthSum = 0.0;
            foreach (var measurement in measurements.Where(m => m != null))
            {
                lengthSum += measurement.Length;
            }

            return await Task.FromResult(new Dictionary<string, object>
            {
                ["lengthSum"] = lengthSum
            });
        }
    }
}